packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -21,79 +21,46 @@ export interface IPersistedFileCache {
     removeEntries(): Promise<void>;
 }
 
-/**
- * Handles garbage collection of expiring cache entries.
- * Not exported.
- * (Based off of the same class in promiseCache.ts, could be consolidated)
- */
-class GarbageCollector<TKey> {
-    private readonly gcTimeouts = new Map<TKey, ReturnType<typeof setTimeout>>();
-
-    constructor(
-        private readonly cleanup: (key: TKey) => void,
-    ) { }
-
-    /**
-     * Schedule GC for the given key, as applicable
-     */
-    public schedule(key: TKey, durationMs: number) {
-        this.gcTimeouts.set(
-            key,
-            setTimeout(
-                () => { this.cleanup(key); this.cancel(key); },
-                durationMs,
-            ),
-        );
-    }
-
-    /**
-     * Cancel any pending GC for the given key
-     */
-    public cancel(key: TKey) {
-        const timeout = this.gcTimeouts.get(key);
-        if (timeout !== undefined) {
-            clearTimeout(timeout);
-            this.gcTimeouts.delete(key);
-        }
-    }
-}
-
 /**
  * Default local-only implementation of IPersistedCache,
  * used if no persisted cache is provided by the host
  */
 export class LocalPersistentCache implements IPersistedCache {
-    private readonly cache = new Map<string, any>();
-    private readonly gc = new GarbageCollector<string>((key) => this.cache.delete(key));
+    private readonly pc: PromiseCache<string, any>;
 
-    public constructor(private readonly snapshotExpiryPolicy = 30 * 1000) {}
+    public constructor(private readonly snapshotExpiryPolicy = 30 * 1000) {
+        this.pc = new PromiseCache<string, any>({ expiry: {policy:"sliding", durationMs: this.snapshotExpiryPolicy} });
```

Comment:
This is tricky.  I don't see a way to preserve existing timeout semantics.  With sliding policy, PromiseCache resets the timeout on read as well as write.  With absolute policy, PromiseCache never resets the policy, not even on write.  But the existing behavior is to reset the timeout on write only.

Reply:
I see - What if I change to absolute ? It would not update on read. Isn't it ?

Reply:
Unless I'm misreading the code, for Absolute policy it _never_ resets the timers, not even on write.

We could discuss adding a new policy type, think it over and see how that would work out.

Reply:
Yes, perhaps we could add a new policy type that would simply call 
        this.gc.cancel(key);
        this.gc.schedule(key, this.snapshotExpiryPolicy);

When the entry is being updated. 
Also, so I can understand it better, do we need to preserve the existing semantics ? When setting to sliding, we do cancel/schedule during gets. Why wasn't it the design ?  Do you know ?